premerge/post-submit-testing.md

```diff
@@ -0,0 +1,202 @@
+`monolithic-windows.sh` scripts) used by the premerge testing. However, there
+will be one main difference between the premerge and postcommit testing
+configurations. In the postcommit configuration we propose testing all projects
+on every commit rather than only testing the projects that themselves changed or
+had dependencies that changed. We propose this for two main reasons. Firstly,
+Buildbot does not have support for heterogenous build configurations. This means
+that testing a different set of projects within a single builder depending upon
+the contents of the commit could easily cause problems. More notifications could
+be produced if certain projects (that were only triggered by some files) were
+failing and some were passing which would significantly increase false positive
+notifications. For example, supposed that we have three commits that land in
+`main` and run through postcommit testing: commit A that touches MLIR, commit B
+that touches clang-tidy, and commit C that modifies MLIR. Commit A lands, then
+commit B, then commit C. If commit A introduces MLIR test failures into an
+otherwise clean slate, we would see the following events:
+
+1. Commit A lands. Because it touches MLIR, the buildbot worker runs the MLIR
+   tests. Some of the tests fail. The buildbot "turns red" and a notification is
+   sent out to the PR author.
+2. Commit B lands. Since it touches clang-tidy, the buildbot worker runs the
+   clang-tidy tests. All of the tests pass. The buildbot "turns green". No
+   notifications are sent out since everything is passing.
+3. Commit C lands. Since it touches MLIR, the buildbot workers runs the MLIR
+   tests. The problem introduced in commit A still exists, so some tests fail.
+   No new tests fail. Since the buildbot was previously green due to the
+   interspersed clang-tidy commit, a notification is still sent out to the
+   author of commit C.
+
+By running the tests for all projects in every postsubmit test run, we avoid
+the problematic situation described above.
+
+Another reason for running all the tests in every postsubmit run: When running
+premerge tests on a PR, we also explicitly do not test certain projects even
+though their dependencies change. While we do this because we suspect
+interactions resulting in test failures would be quite rare, it is possible, and
+having a postcommit configuration catch these rare failures would be useful.
+
+### Data Storage
+
+The hosted Buildbot master instance at [lab.llvm.org](https://lab.llvm.org)
+contains results for all recent postcommit runs. We plan on querying the results
+from the buildbot master because they are already available and that is where
+they will natively be reported after the infrastructure is set up. Buildbot
+supports a [REST API](https://docs.buildbot.net/latest/developer/rest.html) that
+would allow for easily querying the state of a commit in `main`.
+
+For the proposed premerge advisor that tells the user what tests/build failures
```

Comment:
Nit: this refers to a "proposed premerge advisor" but that's the first time this is mentioned in the doc, and the casual reader having no context (me) will be lost. Is there a link to some proposal that should be added here?

This section feel like a bit out-of-place for a document describing the "post-merge testing" flow IMO.

Reply:
It's mentioned briefly in the last paragraph of the background/motivation section.

There is no proposal yet, but I'm hoping to write an RFC/document soon to share with the community once I've fleshed out the details.